pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -516,7 +542,7 @@ fi
 # If we're not just building, then make the new configuration the boot
 # default and/or activate it now.
 if [[ "$action" = switch || "$action" = boot || "$action" = test || "$action" = dry-activate ]]; then
-    if ! targetHostCmd "$pathToConfig/bin/switch-to-configuration" "$action"; then
+    if ! runCmd targetHostCmd "$pathToConfig/bin/switch-to-configuration" "$action"; then
```

Comment:
Will this actually work? I am not seeing you return the error code here: https://github.com/NixOS/nixpkgs/pull/163346/files#diff-fc23c722779993fcffc932965cd143b265aa196d577ca10173b63e5bdcbc0001R161

Reply:
It should work just fine:

```
# foo() { "$@"; }
# foo false
# $?
1: command not found
```